src/Package/Fetch.zig

```diff
@@ -1032,7 +1064,7 @@ fn unpackResource(
     resource: *Resource,
     uri_path: []const u8,
     tmp_directory: Cache.Directory,
-) RunError!void {
+) RunError!?[]const u8 {
```

Comment:
:+1:  Enhancing this function to support returning a sub-path inside the temporary directory makes sense because otherwise the tar unpacking code would need to sometimes move all the top-level entries.

It took me a while to understand the meaning of this return value. Let's make it easier for the next person by adding a doc comment to the function to explain what return value means.

Something like this:

```zig
/// A `null` return value indicates the `tmp_directory` is populated directly with the package contents.
/// A non-null return value means that the package contents are inside a sub-directory indicated by the named path.
```